conans/model/scm.py

```diff
@@ -43,10 +43,11 @@ def recipe_revision(self):
         raise ConanException("Not implemented recipe revision for %s" % self.type)
 
     def __repr__(self):
+        invalid_values = [None, "", {}, []]  # Discard not used/empty values but keep the False ones
```

Comment:
I don't see the reason of this. This fields are: 

```python
self.type = data.get("type")
            self.url = data.get("url")
            self.revision = data.get("revision")
            self.verify_ssl = data.get("verify_ssl")
            self.username = data.get("username")
            self.password = data.get("password")
            self.subfolder = data.get("subfolder", "")
            self.submodule = data.get("submodule")
```

I don't see lists or dicts there. I think I would just check  for ``if v is not None`` and any other value would make it to the final result, why would an empty string be removed (and later when needed it will be None)?

Reply:
Done!